### ISSUE TITLE
Successor #3939: finish fresh-worker config comparison in immutable seed batches

### ISSUE THREAD

**Issue by Unjuno** [Unjuno/agent-interface#3958]
## Preserved predecessor

#3939's single formal invocation is STOP_EXECUTION_ENVELOPE / INCOMPLETE_DENOMINATOR. The outer execution tool terminated before 15 rows; 9 worker exit-0 rows, two references and all partial bytes are retained. Missing final MANIFEST causes the unchanged auditor to exit1. No same-ID continuation/retry or pooling is allowed. Its nine observed rows did not expose a numerical difference; this is not hidden or used to tune the gates. Original #3911 scientific FAIL remains unchanged.

## H / T / D / C / U

**H:** Same as #3939: fresh exec workers do not inherit parent-local thread/determinism settings; explicitly configured 1-thread/deterministic workers match the reference exactly. The registered secondary thread explanation requires BOTH 1-thread modes exact and a multithread logit difference with full state/predictions unchanged. Absence of contrast is a useful HOLD, not a forced positive.

**T:** Change only execution/serialization granularity and allocation identity. Three immutable seed batches replace one monolithic orchestration. Fresh seeds3912201,3912202,3912203; never reuse3912101-3912103 or historical #3911 seeds. One no-retry invocation per seed; aggregate only after all three exit0 and complete manifests. Same exact predecessor blob69a9266981d11eaffc7dc84349356a183b77207b, 512-row/400-update base, rank2, eight first-arrival updates,4096 heldout rows, five arms DEFAULT/T1_D0/T1_D1/T5_D0/T5_D1. Same partial cyclic rotations0/1/2, not fully balanced; timing remains descriptive. Each fresh worker consumes identical sealed JSON and invokes the unmodified predecessor CLI worker. Retain full arrays, states, settings, subprocess exits, import/update/process timings and hashes. Source/gates frozen before the first of three invocations; zero tuning/reruns. Construction performs source/schema/config checks only, no optimizer training.

Environment: provided Linux x86_64 execution container, CPython3.13.5, PyTorch2.10.0+cpu, default threads5/interop5/determinismfalse. No Docker/Podman, GPU, installs, model/provider, network experiment, GUI or input. This is not historical Docker/PyTorch2.5.1 replication.

Additive path `research/system1/needle_resume_worker_config_batches_v2/`; branch `research/issue-3939-worker-config-batches-20260922-v2`. Main/source/branch/path checks precede freeze. One batch serializes before the next tool invocation; if a batch is incomplete, stop without replacement. Each subprocess has a40s safety timeout; outer tool invocation is bounded. Aggregate is raw-only, no training.

**D:** Preserve #3939 gates exactly on all15 new worker rows/3 new references. All three T1_D1 full states/float32 logits/predictions must be exact; a complete same-config mismatch is FAIL_SAME_CONFIG_EQUIVALENCE. Both one-thread arms exact plus a multithread logit difference with unchanged state/predictions supports PASS_WORKER_CONFIG_THREAD_BOUNDARY_SCOPED. No multithread difference is HOLD_THREAD_MECHANISM_NOT_EXPOSED even when H1/H2 pass. Every identity, seed/order, source/freeze/raw hash, process exit, complete denominator and six auditor corruption controls must reconcile. A missing batch is STOP_INCOMPLETE_BATCHES, never partial pooled PASS. Timings have three samples/arm and cannot establish p95<=60ms, a causal historical latency repair or performance benefit.

**C:** Old source/evidence untouched; no shared runtime edits; parent/worker settings are the only experimental factors. Internal predecessor allocation string is retained solely for its unchanged schema validator, while the wrapper/new seeds identify this successor. Independent stdlib audit checks raw states, bitwise float32 logits, labels/argmax, config and provenance, not independent PyTorch PRNG/optimizer trajectories.

**U:** First-arrival synthetic experiment only; no whole16-arrival equivalence, historical causal explanation, model quality, GUI utility, production safety or cross-version generality. A HOLD may narrow the explanation rather than rescue #3911. Roadmap: source-only construction -> batch source freeze -> three first-outcome batches -> independent complete raw audit -> lossless evidence/report and additive PR -> main only if delivery/checks support it. Broad project roadmap remains open.

**Comment by Unjuno** [comment]
Preformal batch freeze complete; 0/3 batches invoked. Branch `research/issue-3939-worker-config-batches-20260922-v2` created from `b2457b746a6df06f6536585dfe2ab937aff639f4`. Seven new source/config construction checks passed with training_updates=0, plus the separately retained #3939 construction. The only harness science delta is new seeds; treatment/gates unchanged. Execution is now one seed per immutable output with an explicit raw-only aggregate.

SHA256:
FREEZE.json `5a26a43bc6f68ab7ada8e3547cb1159282a3a928a2ea746e96136aecb25ff74e`
experiment.py `0feb1116fe3bf9e706702b7a3c2f8493259ab3910dc9c16f308c5de6ddfa6fd3`
worker_entry.py `f9eaeb90a85805718d3b244def27f6f98b696baeec839ac05e0ed155588ab6a8`
predecessor_runner.py `ae12bc6b7fee156bc04ea701fa939313d0db5b5c6eb600daa2153662d82e055c`
aggregate.py `ae86cb847ff83de06b46f1cf91be82a22e6a258395443afe62eff576f4874f34`
audit.py `72dae926b4240b4ec2ddc33c250279bc3b13d9da778baee276a3ea8dd3d3e96e`

Exact commands, each once in order; stop on first missing/nonzero/incomplete batch:
```
python /mnt/data/agent_research/research/system1/needle_resume_worker_config_batches_v2/experiment.py --seed 3912201 --output /mnt/data/agent_research/research/system1/needle_resume_worker_config_batches_v2/formal-01/3912201
python /mnt/data/agent_research/research/system1/needle_resume_worker_config_batches_v2/experiment.py --seed 3912202 --output /mnt/data/agent_research/research/system1/needle_resume_worker_config_batches_v2/formal-01/3912202
python /mnt/data/agent_research/research/system1/needle_resume_worker_config_batches_v2/experiment.py --seed 3912203 --output /mnt/data/agent_research/research/system1/needle_resume_worker_config_batches_v2/formal-01/3912203
python /mnt/data/agent_research/research/system1/needle_resume_worker_config_batches_v2/aggregate.py /mnt/data/agent_research/research/system1/needle_resume_worker_config_batches_v2/formal-01
python /mnt/data/agent_research/research/system1/needle_resume_worker_config_batches_v2/audit.py /mnt/data/agent_research/research/system1/needle_resume_worker_config_batches_v2/formal-01 --output /mnt/data/agent_research/research/system1/needle_resume_worker_config_batches_v2/AUDIT.json
```
All three paths absent at freeze. Source SHA checks guard each batch and aggregation. #3939 partial bytes and auditor failure remain separate, never pooled.

**Comment by Unjuno** [comment]
All three immutable batches completed exit0; 15/15 worker rows and3/3 references retained, no retries/tuning. Raw-only aggregate exit0; frozen stdlib auditor exit0 with errors=[], source_errors=[], manifest_errors=[], and all six corruption controls rejected (missing/duplicate worker, changed logits/optimizer/settings/source).

**Scientific disposition: HOLD_THREAD_MECHANISM_NOT_EXPOSED.** H1 is supported: every fresh DEFAULT reports5 threads/deterministicfalse while the parent is1/true. H2 is supported in this first-arrival sample: all three T1_D1 workers exactly match state/logits/predictions. However **all fifteen workers in every arm** match full state and float32 logits bit-for-bit, with zero changed predictions. Thus neither the thread contrast nor deterministic flag reproduces #3911's historical logit mismatch in this PyTorch2.10 CPU environment. Do not present the aligned-arm match as causal proof or a repair of historical #3911. Its FAIL and #3939's STOP remain unchanged.

Original raw manifest SHA256: `33283139e75c3f462da01ffddd6b5c9b2ea4f16f29dac8a6cea713aef23fd5c5`.
Frozen audit output SHA256: `5f04b4183b3ebd2dba78b92897329f689a595cc13a0e54dcdd6be9f201348e3a`.
Initial full lossless tar.xz of this study (145 files /8,615,528 uncompressed bytes; before report/handoff additions):1,234,884bytes, SHA256 `06f757038d563a45520d772175eb1a87abba56aeda7cede2719c7ced79cac1a5`.
Initial full #3939 retained tar.xz:843,632bytes, SHA256 `d4b7738efc0ab2e4b7392eacae08cbeb4d70e2fcad9aaad16b80d23cfc81d3b6`.
These hashes freeze original executed evidence before any publication packaging.

Per-arm update-only milliseconds across the three seeds (descriptive n=3; no tail or speed claim): DEFAULT6.1108/6.8059/10.2559; T1_D0 4.8402/4.2167/4.9002; T1_D1 14.2590/4.9706/5.0228; T5_D0 6.4612/16.0703/5.3290; T5_D1 9.9831/23.3426/5.4511. Fresh-process wall spans2.571–3.968seconds and is distinct from update-only time. No <=60ms acceptance, model utility, 16-arrival or historical causal claim.

Next: publish scoped report, frozen harness, raw manifest/audit and source-bound evidence delivery. No more training allocations are needed to interpret this completed HOLD.

**Comment by Unjuno** [comment]
Delivered to main through PR #4002, merge `1c817de45cf944d144d76e288823b63bc0ec73a4` from exact verified head `baf33ccedbd9183bb03904eb639f8556d7708620`. PR replay-gate run35648003115 completed success before merge; no force update or protection override. The current-goal2026-09-21 evidence-delivery direction was re-read before integration. Latest-main comparison showed12 additive files, zero deletions, no conflicts with49 intervening main commits.

Postmerge main read-back: report blob `aec02a2549215166951fef7fd3278f434a1c7028`; original full audit blob `66b2aab6b2d1a392c0ddc236a8503d7862b4f7cf`; historical runner still `69a9266981d11eaffc7dc84349356a183b77207b` unchanged. All12 published files, including all four capsule parts, had already matched original local Git blob identities. The four-part decoder was separately verified against all238 original tar members, and the unchanged raw auditor again reproduced SHA256 `5f04b4183b3ebd2dba78b92897329f689a595cc13a0e54dcdd6be9f201348e3a` with zero new optimizer updates/workers/formal runs.

Evidence entry point: `research/system1/needle_worker_config_delivery_3939_3958/README.md`. GitHub representation explicitly requires numerical compatibility and original-byte verification; canonical full verbatim archives are provided separately in the conversation handoff, with their previously posted hashes unchanged.

Closing this bounded executed-and-delivered study as completed **with scientific HOLD_THREAD_MECHANISM_NOT_EXPOSED**, not mechanism PASS. H1/H2 component checks do not explain historical #3911; all15 worker states/logits/predictions agree. #3939 STOP, #3911 FAIL, original-engine2.5.1 causality, full16-arrival scope, latency benefit and broad ROADMAP remain unchanged/open as applicable. No further same-environment allocation is warranted by this negative contrast.
